### ISSUE TITLE
OLE2 Issue with Corel Files 

### ISSUE THREAD

**Issue by ross-spencer** [exponential-decay/skeleton-container-test-suite-generator#3]
Problem with both these files, different signature patterns each. 

        <ContainerSignature Id="24100" ContainerType="OLE2">
            <Description>Corel Presentation</Description>
            <Files>
                <File>
                    <Path>PerfectOffice_MAIN</Path>
                    <BinarySignatures>
                        <InternalSignatureCollection>                    
	                        <InternalSignature ID="24100">
	                            <ByteSequence Reference="BOFoffset">
	                                <SubSequence Position="1" SubSeqMinOffset="0" SubSeqMaxOffset="0">
	                                    <Sequence>'ÿWPC'</Sequence>
	                                </SubSequence>
	                            </ByteSequence>
	                            <ByteSequence Reference="BOFoffset">
	                                <SubSequence Position="1" SubSeqMinOffset="8" SubSeqMaxOffset="0">
	                                    <Sequence>0F 0F 02</Sequence>
	                                </SubSequence>
	                            </ByteSequence>
	                        </InternalSignature>
	                    </InternalSignatureCollection>
                    </BinarySignatures>
                </File>
            </Files>
        </ContainerSignature>
        
        <ContainerSignature Id="24200" ContainerType="OLE2">
            <Description>Compound WordPerfect for Windows Document 6 onwards</Description>
            <Files>
                <File>
                    <Path>PerfectOffice_MAIN</Path>
                    <BinarySignatures>
                        <InternalSignatureCollection>                    
	                        <InternalSignature ID="24200">
	                            <ByteSequence Reference="BOFoffset">
	                                <SubSequence Position="1" SubSeqMinOffset="0" SubSeqMaxOffset="0">
	                                    <Sequence>'ÿWPC'</Sequence>
	                                </SubSequence>
	                            </ByteSequence>
	                            <ByteSequence Reference="BOFoffset">
	                                <SubSequence Position="1" SubSeqMinOffset="8" SubSeqMaxOffset="0">
	                                    <Sequence>01 0A 02 02</Sequence>
	                                </SubSequence>
	                            </ByteSequence>
	                        </InternalSignature>
	                    </InternalSignatureCollection>
                    </BinarySignatures>
                </File>
            </Files>
        </ContainerSignature>

[fmt-892-container-signature-id-24200.zip](https://github.com/exponential-decay/skeleton-container-test-suite-generator/files/1247573/fmt-892-container-signature-id-24200.zip)

[fmt-877-container-signature-id-24100.shw.zip](https://github.com/exponential-decay/skeleton-container-test-suite-generator/files/1253373/fmt-877-container-signature-id-24100.shw.zip)

**Comment by ross-spencer** [comment]
@Dclipsham - @richardlehane has completed some similar work verifying the behaviour in Siegfried with the two Skeleton files. It seems to be an issue with the character ÿ [latin small letter y with diaeresis (U+00FF)](http://graphemica.com/%C3%BF)

I have tried with the character replaced with hex FF in Linux (as per Richard's trials to match the Skeleton file) and we get the right result:

![image](https://user-images.githubusercontent.com/1880412/29749250-831f3ca4-8b7b-11e7-8378-6246bfdedfe7.png)

      <Sequence>FF 'WPC'</Sequence>

I have also tried to modify the Skeleton file to have the UTF-8 sequence:

      00 FF 57 50 43 00 00 00 01 0A 02 02

This did not work.

So it seems dependent on whether you're looking for 0xFF which is ÿ in UTF-8 which is specified by the sig file and formally represented as 0x00FF. Or you want 0x98 in ASCII which is also ÿ. 

I suspect the former as per other Wordperfect examples and [Gary Kessler's info](http://www.garykessler.net/library/file_sigs.html)

In which case, I guess we have another limitation in the Container Signature spec when using characters that cross from extended-ASCII into UTF-8.

      <Sequence>FF 'WPC'</Sequence>

Seems to be a good, immediate workaround.
